pkg/config/system_probe.go

```diff
@@ -117,6 +117,7 @@ func InitSystemProbeConfig(cfg Config) {
 
 	cfg.BindEnvAndSetDefault(join(spNS, "max_tracked_connections"), 65536)
 	cfg.BindEnv(join(spNS, "max_closed_connections_buffered"))
+	cfg.BindEnvAndSetDefault(join(spNS, "closed_connection_flush_threshold"), 0)
```

Comment:
We set the default to `0` here, but then later on (https://github.com/DataDog/datadog-agent/pull/14164/files#diff-9c2a90c0163f3424a0f83f16a663c759b5b5b2109f88217eec2b8ed4f22ccc81R436) if it is set to `0`, then we update it to be `threshold/2`.

I think this has the following negative side effects:
- If customer set this to 0 (intentionally), we'd be updating it out from under them.
- The actual value sent to the driver is hidden such that while it appears to be `0`, when sent to the driver it is `threshold/2`.

I think it'd be clearer if we did one of or a combination of the following options:
- Set the default to `threshold/2` always, and then if customer sets it to `0`, log an error because I think _actually_ setting this to `0` in the driver will cause us to thrash.
- If we want to be able to tell if the customer set it to `0`, we could do something like (psuedo-code):
```
if cfg.IsSet(join(spNS, "closed_connection_flush_threshold")) {
    if cfg.closed_connection_flush_threshold == 0 {
        log.error()
        ....
}
}
```

Essentially, since `0` is an invalid configuration value and its possible that it could be intentionally set to this, we should detect that and log / fail out.